### DISCUSSION TITLE
RT60 calculation

### DISCUSSION THREAD

**Discussion by acbof** [Ifsttar/I-Simpa#306] [Q&A] [answered]
Hello,

I am calculating the Reverberation Time on a room and think there's something off with the values found in SPPS simulation. 

Usually, I need to calculate RT60, not 30 or 15, so I setup the Acoustic Parameters for this, as follows: 
![image](https://user-images.githubusercontent.com/29489885/154696199-0ab45091-5328-42d9-8cf5-25510f35e8ea.png)

Getting the following results:
![image](https://user-images.githubusercontent.com/29489885/154696828-84ff9afd-66ab-4adb-8d40-6d63401fb071.png)

How is it that the time required for the sound in a room to decay 60 dB is less than the time for decaying 30 dB? 

Also, I would like to understand what the "NaN" values stand for. 

Thank you!

**Comment by Picaut** [comment, marked as answer]
To evaluate the reverberation time RT60, you need to have a linear decrease of the sound level of more than 60dB (between -5dB from the max sound level to -65 dB] on the Schroeder curve. Most of time the sound level dynamic is not enough and the last point at -65dB from the max is not available. See here for the formula:
https://i-simpa-wiki.readthedocs.io/fr/latest/room_acoustics_parameters.html?highlight=reverberation%20time#reverberation 
In this case, the last point of the curve is considered, but due to the shape of the curve, the value is not correct.
When considering RT30 or RT15, most of time, the dynamic is enough, meaning that you are able to calculate the reverberation time on -15dB or -30dB.
The calculation of RT60 should be difficult most of time. To increase the dynamic, with SPPS, you should increase the number of sound particles (in the Random calculation mode) and/or increase the 'Limit value of the particle extinction: ratio 10^n” parameter : https://i-simpa-wiki.readthedocs.io/fr/latest/tab_calculation.html?highlight=extinction#spps-calculation-parameters
Increasing the number of particle by a factor will increase the dynamic of a max value of 3dB, but increasing considerably the computational time.
I suggest you to consider only RT15 et RT30

Considering the NaN value, it occurs when I-Simpa can not defined the -5 or -35dB point on the Schroeder curve, for example in the case or RT30 calculation. The software returns NaN "Not a Number"

**Comment by Picaut** [comment]
Referring to the figure below, the point A3 need to be find for the RT60 calculation. If A3 is not in linear part of the Schroeder curve, the calculated RT is biased by the end of the curve.
![image](https://user-images.githubusercontent.com/7996615/154726217-71b95606-1f0d-4b06-bb1a-d649bc181e7f.png)

**Reply by acbof** [reply]
What if I increase the simulation time, so the decay can reach 60 dB? Would that work? My simulation is 7 s long, but the values for RT are 8 s or more, so I didn't think it could be that. 

As for considering RT15 and 30, I need the 60 dB value, because that's what my local standard asks for. But even for those values, some results for RT30 are less than those for RT15, which still doesn't make any sense to me.

**Reply by Picaut** [reply]
You have very large RT. So if you want to be able to calculate the RT60, your simulation must be greater than the RT. However, you should have enough dynamics.

Note, that it may be curious to obtained so important RT: do you add material absorption in your model ?

**Reply by acbof** [reply]
I defined some usual construction materials (masonry, glass, gypsum, etc). But it's a very large space (~1000 m³), so it's natural that the RT is this big. Even for Sabine's RT, that is usually smaller, the values are around 6s. 

That is also the reason I was relutant in increasing the simulation time and number of particles. For small values of these, the simulation already takes 30 minutes to complete (I believe is due to the rooms's size).
